### ISSUE TITLE
This profile set might be unnecessary

### ISSUE THREAD

**Issue by szszoke** [szszoke/sennheiser-gsp670-pulseaudio-profile#18]
Hello!

I updated my Manjaro system to the `5.15.2-2-MANJARO` kernel and the dongle with the updated firmware just started working. I did not have this profile installed on my system.

I suppose the patch with the quirks table entry made it into a stable release.

The only out-of-ordinary thing on my system is that I am using PipeWire with Wireplumber as my audo server.

I would like to ask for some community feedback:

1. Is the dongle with the updated firmware working on your system?
2. Are you using `PipeWire` or `PulseAudio`?
3. What Kernel version are you running?
4. Are using this custom profile or not?
5. Is the dongle recognized as two outputs and a microphone?

**Comment by Akselmo** [comment]
Hi!

1. I updated the dongle last year, I cant remember exactly when but it was when I was still using Windows. If there has been any updates after that, I havent updated it further.
2. PulseAudio
3. 5.13.0-21-generic, I still need to manually patch every time the kernel updates
4. I am using following conf, `/etc/udev/rules.d/91-pulseaudio-sennheiser-gsp670.rules`, so I think I am not using the same profile as in repo
5. Yup, theres one mono (chat), one stereo (game) and mic
```
# GSA-70 dongle with updated firmware
ATTRS{idVendor}=="1395", ATTRS{idProduct}=="0300", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
# GSA-70 dongle with older firmware
ATTRS{idVendor}=="1395", ATTRS{idProduct}=="0089", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
# GSP-670 headset
ATTRS{idVendor}=="1395", ATTRS{idProduct}=="008a", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
```

**Comment by szszoke** [comment]
> Hi!
> 
> 1. I updated the dongle last year, I cant remember exactly when but it was when I was still using Windows. If there has been any updates after that, I havent updated it further.
> 2. PulseAudio
> 3. 5.13.0-21-generic, I still need to manually patch every time the kernel updates
> 4. I am using following conf, `/etc/udev/rules.d/91-pulseaudio-sennheiser-gsp670.rules`, so I think I am not using the same profile as in repo
> ```
> # GSA-70 dongle with updated firmware
> ATTRS{idVendor}=="1395", ATTRS{idProduct}=="0300", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
> # GSA-70 dongle with older firmware
> ATTRS{idVendor}=="1395", ATTRS{idProduct}=="0089", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
> # GSP-670 headset
> ATTRS{idVendor}=="1395", ATTRS{idProduct}=="008a", ENV{PULSE_PROFILE_SET}="usb-gaming-headset.conf"
> ```
> 

Hey! Thanks for the reply. I added a fifth question because I realized that I forgot.

**Comment by Akselmo** [comment]
> Hey! Thanks for the reply. I added a fifth question because I realized that I forgot.

updated :)

**Comment by mobergmann** [comment]
Sadly there is no communication channel anymore or am I missing something?

**Comment by szszoke** [comment]
> Sadly there is no communication channel anymore or am I missing something?

Please try to answer the questions from the first post.

**Comment by mobergmann** [comment]
> > Sadly there is no communication channel anymore or am I missing something?
> 
> Please try to answer the questions from the first post.

Oh, I'm sorry.

#### Answers

1. > Is the dongle with the updated firmware working on your system?
    - yes
2. > Are you using PipeWire or PulseAudio?  
    - PulseAudio
3. > What Kernel version are you running?
    - 5.15.4-arch1-1
4. > Are using this custom profile or not?
    - I tried both (same result)
5. > Is the dongle recognized as two outputs and a microphone?
    - only one output and a microphone

**Comment by cottton** [comment]
_First: i have no idea how audio on linux works.
I just would love to get this think working again._

I recently bought this headset.
Used it on ubuntu 18.04 and had no stereo sound.
Came here and used this - worked.

Was an id**t and installed the new firmware (didnt know about that probelm)
and now got no mic on wireless and massive problems when switching to cable.

1. Is the dongle with the updated firmware working on your system?
yes
2. Are you using PipeWire or PulseAudio?
I dont know.
3. What Kernel version are you running?
I dont know.
4. Are using this custom profile or not?
Yes
5. Is the dongle recognized as two outputs and a microphone?
No (i think?)

---

Is there a way to roll this profile here back to see if its working without?

**Comment by szszoke** [comment]
> _First: i have no idea how audio on linux works. I just would love to get this think working again._
> 
> I recently bought this headset. Used it on ubuntu 18.04 and had no stereo sound. Came here and used this - worked.
> 
> Was an id**t and installed the new firmware (didnt know about that probelm) and now got no mic on wireless and massive problems when switching to cable.
> 
>     1. Is the dongle with the updated firmware working on your system?
>        yes
> 
>     2. Are you using PipeWire or PulseAudio?
>        I dont know.
> 
>     3. What Kernel version are you running?
>        I dont know.
> 
>     4. Are using this custom profile or not?
>        Yes
> 
>     5. Is the dongle recognized as two outputs and a microphone?
>        No (i think?)
> 
> 
> Is there a way to roll this profile here back to see if its working without?

If you installed via the DEB package, you should be able to uninstall via `apt`.

```
sudo apt remove pulseaudio-sennheiser-gsp670
```

**Comment by etlam262** [comment]
> 1. Is the dongle with the updated firmware working on your system?

Yes

> 2.  Are you using PipeWire or PulseAudio?

PulseAudio

> 3.  What Kernel version are you running?

Linux 5.10.96-1-MANJARO x86_64

> 4.  Are using this custom profile or not?

Yes

> 5. Is the dongle recognized as two outputs and a microphone?

It was until two days ago. Unfortunately I was using Windows and had some issues with the microphone not working. I saw that there was an update available in the EPOS software and installed the new firmware version 3.2.6. In Windows everything works fine again but now on Manjaro the device is shown as "Analog Output - EPOS GSA 70" and only the mono channel is available (see screenshot). Interestingly enough it says "Profile: Analog Stereo Output" when I plug the dongle into my Notebook (where the profile is installed as well) but I can't choose a different profile either.

I also tried to remove the profile but that didn't change anything.
I suppose I have to change some names in the config files to make it work again and was wondering if you can tell me where I have to make adjustments and what I have to do.

![Screenshot from 2022-02-13 19-12-17](https://user-images.githubusercontent.com/45878768/153768829-42aa906a-81d3-49ae-9923-a72375837f99.png)

**Comment by szszoke** [comment]
@etlam262 you have made an irreversible change to your dongle by upgrading the firmware and your only shot to use it with Linux is to either patch your current kernel with a USB Audio quirk or update to 5.15, which should contain the quirk.

There is also a second branch in this repository that contains changes for the new firmware version. You have to use that or switch to PipeWire.

PipeWire seems to have a profile set built in so assuming that your kernel is patched or recent, you don't need this profile set anymore.

**Comment by etlam262** [comment]
Thanks for the quick reply. Are there any downsides to updating the kernel to 5.15.21-1 (via the Manjaro Settings Manager)? I'd have thought that Manjaro beeing a sort of rolling release distro it would keep the kernel up to date itself.

**Comment by etlam262** [comment]
I finally came around to do it and upgraded the Kernel to 5.15 (and also tried 5.16) but unfortunately that didn't fix the issue. Then I tried to install the profiles from both branches. After installing the profile from the master branch nothing changed and after installing the profile from the "new-dongle-firmware"-branch the dongle wasn't recognized as audio device anymore.

EDIT:
It took some effort but installing pipewire fixed the problem. Now everything is working.
